samples/dagger-hilt/library/src/main/kotlin/com/bumble/appyx/sample/hilt/library/externaldeps/ExternalDepsLibraryCustomNodeFactory.kt

```diff
@@ -0,0 +1,39 @@
+package com.bumble.appyx.sample.hilt.library.externaldeps
+
+import com.bumble.appyx.core.modality.BuildContext
+import com.bumble.appyx.core.node.Node
+import com.bumble.appyx.dagger.hilt.CustomNodeFactory
+import com.bumble.appyx.dagger.hilt.HiltAppyxCustomNodeFactory
+import com.bumble.appyx.dagger.hilt.HiltNodeFactoryProvider
+import com.bumble.appyx.dagger.hilt.getNodeFactory
+import com.bumble.appyx.sample.hilt.library.InternalDependency
+import javax.inject.Inject
+
+/**
+ * This is the only kotlin type that needs to be public.
+ * Dagger hilt will correctly bind the actual factory (below) to this interface.
+ */
+interface ExternalDepsLibraryCustomNodeFactory : CustomNodeFactory {
+    fun create(buildContext: BuildContext, hiltNodeFactoryProvider: HiltNodeFactoryProvider): Node
+}
+
+/**
+ * This @HiltAppyxCustomNodeFactory annotation causes a dagger module
+ * (with ActivityComponent scope) to be created.
+ *
+ * This module uses a Provider to ensure that the factory is only instantiated when required.
+ */
+@HiltAppyxCustomNodeFactory(ExternalDepsLibraryCustomNodeFactory::class)
+internal class ExternalDepsLibraryCustomNodeFactoryImpl @Inject constructor(
+    private val internalDependency: InternalDependency
+) : ExternalDepsLibraryCustomNodeFactory {
+    override fun create(
+        buildContext: BuildContext,
+        hiltNodeFactoryProvider: HiltNodeFactoryProvider
+    ) =
+        ExternalDepsLibraryNodeImpl(
+            buildContext = buildContext,
+            noDepsLibraryNodeFactory = hiltNodeFactoryProvider.getNodeFactory(),
+            dependency = internalDependency
+        )
+}
```

Comment:
You don't need to provide a factory for every node, just nodes that take dependencies. Take a look at the `NoDepsLibraryNode` for example.

You also don't need to pass `hiltNodeFactoryProvider` to all nodes, just nodes that might want to create a node using this approach, it's optional and just for demonstration purposes.

Reply:
so if you were using Appyx as the core navigational framework for your whole application you'd be needing dependencies pretty commonly. To satisfy that you'd need a lot of factories to propagate the hiltNodeFactoryProvider throughout the nav graph, right?

I'm just wondering what this would look like in practice in a non-trivial app or at-scale in general and wondering if there's something we can do to avoid boilerplate dependency provisioning in the common usecase of providing a set of dependencies to a node that may be nested arbitrarily deep in a navigation graph.